### DISCUSSION TITLE
Restore Borg-Backup on another Machine

### DISCUSSION THREAD

**Discussion by mighty-services** [nextcloud/all-in-one#4106] [Questions] [answered]
### Steps to reproduce
1. Backup your current AIO-Instance 
![image](https://github.com/nextcloud/all-in-one/assets/71066192/452ad8f5-4aab-4af1-bc89-973d88c2a3ca)

3. copy the borg-backup folder to the new instance with rsync by issuing this line:
`rsync -AaP /backup/borg <root-user>@<IP-Address>:/backup/borg`

4. Setup empty AIO Instance on another VM

5. In the Mastercontainer-Config, point to the Backup-Folder and paste in the secret Key of the old BorgBackup-Config
![image](https://github.com/nextcloud/all-in-one/assets/71066192/83db34c2-7f62-43fd-a763-1ab5df1fc929)

6. Start the recovery-process

### Expected behavior
Restore the copied Backup into a new AIO-Instance.

### Actual behavior
The Recovery Process stops after a few second with this error message
```
Local Exception
Traceback (most recent call last):
  File "/usr/lib/python3.11/site-packages/borg/archiver.py", line 5343, in main
    exit_code = archiver.run(args)
                ^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/borg/archiver.py", line 5263, in run
    return set_ec(func(args))
                  ^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/borg/archiver.py", line 176, in wrapper
    kwargs['manifest'], kwargs['key'] = Manifest.load(repository, compatibility)
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/borg/helpers/manifest.py", line 190, in load
    data = key.decrypt(None, cdata)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/borg/crypto/key.py", line 452, in decrypt
    payload = self.cipher.decrypt(data)
              ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "src/borg/crypto/low_level.pyx", line 311, in borg.crypto.low_level.AES256_CTR_BASE.decrypt
  File "src/borg/crypto/low_level.pyx", line 428, in borg.crypto.low_level.AES256_CTR_BLAKE2b.mac_verify
borg.crypto.low_level.IntegrityError: MAC Authentication failed

Platform: Linux ebf839aa7261 5.15.0-91-generic #101-Ubuntu SMP Tue Nov 14 13:30:08 UTC 2023 x86_64
Linux: Unknown Linux  
Borg: 1.2.7  Python: CPython 3.11.6 msgpack: 1.0.5 fuse: llfuse 1.4.2 [pyfuse3,llfuse]
PID: 25  CWD: /
sys.argv: ['/usr/bin/borg', 'list', '/mnt/borgbackup/borg']
SSH_ORIGINAL_COMMAND: None

The entered path seems to be valid but could not open the backup archive.
Most likely the entered password was wrong so please adjust it accordingly!
```

### Host OS 
Both instances are Ubuntu Server 22.04lts 

docker version on the old VM
```
Client:
 Version:           24.0.5
 API version:       1.43
 Go version:        go1.20.3
 Git commit:        24.0.5-0ubuntu1~22.04.1
 Built:             Mon Aug 21 19:50:14 2023
 OS/Arch:           linux/amd64
 Context:           default
```

docker version on the new VM
```
Client: Docker Engine - Community
 Version:           25.0.0
 API version:       1.44
 Go version:        go1.21.6
 Git commit:        e758fe5
 Built:             Thu Jan 18 17:09:49 2024
 OS/Arch:           linux/amd64
 Context:           default
```

#### Nextcloud AIO version and channel
both master containers are on the latest stable release 7.10.0

#### Other valuable info 
The master conteriner ist running with the docker-compose template from this github-page.
Since I´m using a reverse proxy, I uncommented the Apache Port and Bind-Address environment variable and forwarded it on my NGINX Reverse Proxy Manager. This is the docker-compose file in my new instance, created in Portainer:

```
services:
  nextcloud-aio-mastercontainer:
    image: nextcloud/all-in-one:latest
    init: true
    restart: always
    container_name: nextcloud-aio-mastercontainer 
    volumes:
      - nextcloud_aio_mastercontainer:/mnt/docker-aio-config
      - /var/run/docker.sock:/var/run/docker.sock:ro 
    ports:
      - 80:80 
      - 8080:8080
      - 8443:8443
    environment: 
      # - AIO_DISABLE_BACKUP_SECTION=false 
      - APACHE_PORT=11000 
      - APACHE_IP_BINDING=0.0.0.0 
      - BORG_RETENTION_POLICY=--keep-within=7d --keep-weekly=4 --keep-monthly=6 
      # - COLLABORA_SECCOMP_DISABLED=false
      - NEXTCLOUD_DATADIR=/data/ncdata 
      - NEXTCLOUD_MOUNT=/mnt/ 
      - NEXTCLOUD_UPLOAD_LIMIT=10G 
      - NEXTCLOUD_MAX_TIME=3600 
      - NEXTCLOUD_MEMORY_LIMIT=512M 
      # - NEXTCLOUD_TRUSTED_CACERTS_DIR=/path/to/my/cacerts 
      # - NEXTCLOUD_STARTUP_APPS=deck twofactor_totp tasks calendar contacts notes 
      - NEXTCLOUD_ADDITIONAL_APKS=imagemagick 
      - NEXTCLOUD_ADDITIONAL_PHP_EXTENSIONS=imagick
      # - NEXTCLOUD_ENABLE_DRI_DEVICE=true 
      # - NEXTCLOUD_KEEP_DISABLED_APPS=false 
      # - TALK_PORT=3478
      # - WATCHTOWER_DOCKER_SOCKET_PATH=/var/run/docker.sock
    # networks:
      # - nextcloud-aio 


volumes:
  nextcloud_aio_mastercontainer:
    name: nextcloud_aio_mastercontainer
```

**Comment by szaimen** [comment, marked as answer]
> Local Exception
> Traceback (most recent call last):
>   File "/usr/lib/python3.11/site-packages/borg/archiver.py", line 5343, in main
>     exit_code = archiver.run(args)
>                 ^^^^^^^^^^^^^^^^^^
>   File "/usr/lib/python3.11/site-packages/borg/archiver.py", line 5263, in run
>     return set_ec(func(args))
>                   ^^^^^^^^^^
>   File "/usr/lib/python3.11/site-packages/borg/archiver.py", line 176, in wrapper
>     kwargs['manifest'], kwargs['key'] = Manifest.load(repository, compatibility)
>                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/lib/python3.11/site-packages/borg/helpers/manifest.py", line 190, in load
>     data = key.decrypt(None, cdata)
>            ^^^^^^^^^^^^^^^^^^^^^^^^
>   File "/usr/lib/python3.11/site-packages/borg/crypto/key.py", line 452, in decrypt
>     payload = self.cipher.decrypt(data)
>               ^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "src/borg/crypto/low_level.pyx", line 311, in borg.crypto.low_level.AES256_CTR_BASE.decrypt
>   File "src/borg/crypto/low_level.pyx", line 428, in borg.crypto.low_level.AES256_CTR_BLAKE2b.mac_verify
> borg.crypto.low_level.IntegrityError: MAC Authentication failed
> 
> Platform: Linux ebf839aa7261 5.15.0-91-generic #101-Ubuntu SMP Tue Nov 14 13:30:08 UTC 2023 x86_64
> Linux: Unknown Linux  
> Borg: 1.2.7  Python: CPython 3.11.6 msgpack: 1.0.5 fuse: llfuse 1.4.2 [pyfuse3,llfuse]
> PID: 25  CWD: /
> sys.argv: ['/usr/bin/borg', 'list', '/mnt/borgbackup/borg']
> SSH_ORIGINAL_COMMAND: None

Hi, this looks like a bug in borg. Feel free to create an issue here: https://github.com/borgbackup/borg/issues

**Comment by Kirk1984** [comment]
Shouldn't /backup be a NEXTCLOUD_MOUNT in the new compose? How could AIO get /backup/borg when /mnt is made avaiable for AIO?
